floris/tools/optimization/layout_optimization/layout_optimization_scipy.py

```diff
@@ -0,0 +1,316 @@
+# Copyright 2022 NREL
+
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not
+# use this file except in compliance with the License. You may obtain a copy of
+# the License at http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations under
+# the License.
+
+# See https://floris.readthedocs.io for documentation
+
+import numpy as np
+import matplotlib.pyplot as plt
+from scipy.optimize import minimize
+from shapely.geometry import Point
+from scipy.spatial.distance import cdist
+
+from .layout_optimization_base import LayoutOptimization
+
+class LayoutOptimizationScipy(LayoutOptimization):
+    def __init__(
+        self,
+        fi,
+        boundaries,
+        freq=None,
+        x0=None,
+        bnds=None,
+        min_dist=None,
+        solver='SLSQP',
+        optOptions=None,
+    ):
+        super().__init__(fi, boundaries, min_dist=min_dist, freq=freq)
+
+        if optOptions is None:
+            self.optOptions = {"maxiter": 100, "disp": True, "iprint": 2, "ftol": 1e-9, "eps":0.01}
+
+        self.reinitialize_opt(
+            boundaries=boundaries,
+            freq=freq,
+            x0=x0,
+            bnds=bnds,
+            solver=solver,
+            optOptions=self.optOptions,
+        )
+
+
+    # Private methods
+
+    def _optimize(self):
+        self.residual_plant = minimize(
+            self._obj_func,
+            self.x0,
+            method=self.solver,
+            bounds=self.bnds,
+            constraints=self.cons,
+            options=self.optOptions,
+        )
+
+        return self.residual_plant.x
+
+    def _obj_func(self, locs):
```

Comment:
Is the objective function different per optimizer? Could (part of) it be moved to the base class?

Reply:
The objectives are not different in terms of the value they compute, but the format of design variables and what is returned by the objective function is different for pyoptsparse and scipy, so that is the reason for them being specified within the optimizer classes.